examples/cluster/plot_kmeans_silhouette_analysis.py

```diff
@@ -0,0 +1,128 @@
+                  center_box=(-10.0, 10.0),
+                  shuffle=True,
+                  random_state=1)  # For reproducibility
+
+range_n_clusters = [2, 4, 6]
+
+for n_clusters in range_n_clusters:
+    # Create a subplot with 1 row and 2 columns
+    fig, (ax1, ax2) = plt.subplots(1, 2)
+    fig.set_size_inches(18, 7)
+
+    # The 1st subplot is the silhouette plot
+    # The silhouette coefficient can range from -1, 1 but in this example all
+    # lie within [-0.1, 1]
+    ax1.set_xlim([-0.1, 1])
+    # The (n_clusters+1)*10 are the additional samples to demarcate the space
+    # between silhouette plots of individual clusters.
+    ax1.set_ylim([0, len(X) + (n_clusters + 1) * 10])
+
+    # Initialize the clusterer with n_clusters value and a random generator
+    # seed of 10 for reproducibility.
+    clusterer = KMeans(n_clusters=n_clusters, random_state=10)
+    cluster_labels = clusterer.fit_predict(X)
+
+    # The silhouette_score gives the average value for all the samples.
+    # This gives a perspective into the density and separation of the formed
+    # clusters
+    silhouette_avg = silhouette_score(X, cluster_labels)
+    print("For n_clusters =", n_clusters,
+          "The average silhouette_score is :", silhouette_avg)
+
+    # Compute the silhouette scores for each sample
+    sample_silhouette_values = silhouette_samples(X, cluster_labels)
+
+    y_lower = 10
+    for i in range(n_clusters):
+        # Aggregate the silhouette scores for samples belonging to
+        # cluster i, and sort them
+        ith_cluster_silhouette_values = \
+            sample_silhouette_values[cluster_labels == i]
+
+        ith_cluster_silhouette_values.sort()
+
+        size_cluster_i = ith_cluster_silhouette_values.shape[0] 
+        y_upper = y_lower + size_cluster_i
+
+        color = cm.spectral(float(i) / n_clusters, 1)
+        ax1.fill_betweenx(np.arange(y_lower, y_upper),
+                          0, ith_cluster_silhouette_values,
+                          facecolor=color, edgecolor=color)
+
+        # Label the silhouette plots with their cluster numbers at the middle
+        ax1.text(-0.05, y_lower + 0.5 * size_cluster_i, str(i))
+
+        # Compute the new y_lower for next plot
+        y_lower = y_upper + 10  # 10 for the 0 samples
+
+    ax1.set_title("The silhouette plot for the various clusters.")
+    ax1.set_xlabel("The silhouette coefficient values")
+    ax1.set_ylabel("Cluster label")
+    
+    # The vertical line for average silhoutte score of all the values
+    ax1.axvline(x = silhouette_avg, color = "red", linestyle = "--")
+
+    ax1.set_yticks([])  # Clear the yaxis labels / ticks
+    ax1.set_xticks([-0.1, 0, 0.2, 0.4, 0.6, 0.8, 1])
+
+    # 2nd Plot showing the actual clusters formed
+
+    cmap_samples = [cm.spectral(float(cluster_labels[i]) / n_clusters)
+                    for i in range(X.shape[0])]
+    ax2.scatter(X[:, 0], X[:, 1], marker='.', s=30, lw=0,
+                c=cmap_samples)
```

Comment:
Could I leave this as such... I am unable to reproduce the color easily otherwise... 

(seems like the the integer needs to be normed before `plt.get_cmap()` can convert it... I have tried normalizing it using `float(i) / n_clusters` but the colors produced for same i by `scatter` and `plt.get_cmap()(float(i) / n_clusters)` are different. I've tried using different cmaps too.. like cm.jet / cm.spectral... Perhaps you might have a solution though? )